### PR TITLE
Minor changes from arrays work

### PR DESCRIPTION
**Pull request by mppf**
A grab-bag of minor changes from my arrays branch.

* avoid using unusual Chapel style (a let expression) in chpl__extendedEuclid when regular variables will do
* adjust CyclicDist for unsigned idxType
* in chpl__buildArrayExpr, remove a :?t that is unused but which I think doesn't belong
* mark chpl__autoDestroy functions in ChapelBase with that pragma to avoid problems in needing to know the flag before resolution adds it
* rename alignedActuals/alignedFormals to match what they actually mean (now called formalIdxToActual and actualIdxToFormal).
* adjust computeActualFormalAlignment to avoid uselessly checking a formal we've already aligned

Passes quickstart testing and gasnet testing over release/examples and distributions.
Passed full local testing.
Reviewed by @benharsh - thanks!

### CODE REVIEW

File: test/release/examples/benchmarks/shootout/meteor-fast.chpl [hunk 1]
```diff
@@ -269,13 +269,14 @@ proc searchParallel(in board=0, in pos=0, used=0, placed=0, firstPiece=0) {
       if allMasks[currentMask] {
         const mask = allMasks[currentMask];
 
-        const currentSolution: [piecesDom] int =
-          [firstPiece, mask, 0, 0, 0, 0, 0, 0, 0, 0];
 
-        begin with (in board, in pos)
+        begin with (in board, in pos) {
+          const currentSolution: [piecesDom] int =
+            [firstPiece, mask, 0, 0, 0, 0, 0, 0, 0, 0];
```
Comment (benharsh) [inline test/release/examples/benchmarks/shootout/meteor-fast.chpl]: @ben-albrecht or @lydia-duncan , does this change look OK to you?

**Comment by benharsh** [comment]
The autoDestroy flag change seems unfortunate, but the rest looks reasonable to me. I'd want to check with either ben or lydia on the meteor change, just to make sure they're at least aware of it. 👍

File: test/release/examples/benchmarks/shootout/meteor-fast.chpl (hunk 1 above)
Comment (mppf) [inline test/release/examples/benchmarks/shootout/meteor-fast.chpl]: The issue is that according to our specified array semantics, the local variable currentSolution can be destroyed (when the function exits) before the begin runs. There may be some algorithm reason that is not possible (that I have not noticed) though.

Comment (lydia-duncan) [inline test/release/examples/benchmarks/shootout/meteor-fast.chpl]: I thought that our solution to handling that was to enclose the declaration of the variable being used and the begin statement in a sync statement to guarantee it did not get referenced outside of its scope.

Moving the variable to be local within the scope of the begin is fine with me, too.  Did you notice a performance impact for this change?

Comment (mppf) [inline test/release/examples/benchmarks/shootout/meteor-fast.chpl]: I ran it twice in each configuration
* 0.03845, 0.04365 without this change
* 0.03531, 0.04106 with this change

That looks "in the noise" to me but might be a slight benefit.

Comment (lydia-duncan) [inline test/release/examples/benchmarks/shootout/meteor-fast.chpl]: That's good enough for me!

Comment (ben-albrecht) [inline test/release/examples/benchmarks/shootout/meteor-fast.chpl]: +1

**Pull request merged**